File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.Brand;

import java.util.List;

/**
 * 品牌接口
 * @author mofei
 * @date 2018/9/1 22:33
 */
public interface BrandService {


    List<Brand> listBrand();

    /**
     *
     * 功能描述: 品牌分页
     *
     * @param: [pageNum, rows]
     * @return: PageResult
     * @auther: mofei
     * @date: 2018/9/5 10:16
     */
    PageResult findPage(Brand brand,int pageNum, int pageSize);
    /**
     *
     * 功能描述: 品牌分页
     *
     * @param: [pageNum, rows]
     * @return: PageResult
     * @auther: mofei
     * @date: 2018/9/5 10:16
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加品牌
     * @param brand 品牌信息
     */
    void addBrand(Brand brand);

    /**
     * 修改品牌
     * @param brand 品牌信息
     */
    void updateBrand(Brand brand);

    /**
     * 根据ID查找品牌
     * @param id
     * @return
     */
    Brand findOne(Long id);

    /**
     * 删除
     * @param id 编号
     */
    void deleteBrand(Long[] id);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌的服务层
app.service('brandService',function ($http) {
    this.findPage=function (page,rows) {
        return  $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }

    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&&rows="+rows,searchEntity);
    }

    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    this.findOne= function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    this.delete = function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }

    //品牌下来
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }
});